components/com_fabrik/models/form.php

```diff
@@ -2333,14 +2335,18 @@ public function validate()
 			$post[$key] = $val;
 		}
 
-		if (!empty($this->errors))
+		// (Bauer) This would always be run because $this-errors contains an array with keys for each element - so $this->errors will never be an empty array. 
+		// Isn't that why the $ok flag was created to begin with???
+		// if (!empty($this->errors))
+		if(!$ok)
 		{
 			FabrikWorker::getPluginManager()->runPlugins('onError', $this);
-		}
-
-		FabrikHelperHTML::debug($this->errors, 'form:errors');
-		//echo "<pre>";print_r($this->errors);exit;
-		$this->setErrors($this->errors);
+			FabrikHelperHTML::debug($this->errors, 'form:errors');
+			//echo "<pre>";print_r($this->errors);exit;
+		
+			// (Bauer) Does this even need to be done if $ok is true? 
+			$this->setErrors($this->errors);
```

Comment:
OK. 
But checking for if(!empty($this->errors)) is still not what should be checked here, no? It should be to checking for If(!$ok). Like I said, there's a reason that $ok flag was created to begin with. If left as is, this will continue to allow the FabrikWorker::getPluginManager()->runPlugins('onError', $this) function to be run even when it does not need to be run.